<launch>
  <arg name="results_path" default="/home/jaeyoung/data/local_benchmark/local_benchmark.csv" />
  <arg name="trajectory_path" default="/home/jaeyoung/data/local_benchmark/local_trajectory.csv" />

  <arg name="frame_id" default="world" />

  <node name="hilbert_local_planning_benchmark" pkg="hilbert_planning_benchmark" type="local_hilbert_planning_benchmark_node" output="screen" args="--alsologtostderr">
    <param name="results_path" value="$(arg results_path)" />
    <param name="trajectory_output_path" value="$(arg trajectory_path)"/>
    <param name="visualize" value="true" />
    <param name="robot_radius" value="0.5" />
    <param name="publish_tsdf_map" value="true" />
    <param name="publish_esdf_map" value="true" />
    <param name="tsdf_voxel_size" value="0.20" />
    <param name="tsdf_voxels_per_side" value="16" />
    <param name="clear_sphere_radius" value="1.5" />
    <param name="occupied_sphere_radius" value="4.0" />
    <param name="update_mesh_every_n_sec" value="0.0" />
    <param name="num_seconds_to_plan" value="5.0" />
    <param name="camera_max_dist" value="20.0" />
    <param name="max_ray_length_m" value="5.0" />
    <param name="frame_id" value="$(arg frame_id)" />
    <param name="world_frame" value="$(arg frame_id)" />
    <param name="publish_pointclouds" value="true" />
    <param name="verbose" value="true" />
    <param name="hilbertmap/width" value="7.0"/>
    <param name="hilbertmap/length" value="7.0"/>
    <param name="hilbertmap/height" value="3.0"/>
    <param name="hilbertmap/resolution" value="0.5"/>
  </node>
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find hilbert_planning_benchmark)/launch/config.rviz" />

</launch>
